File: ITI_Angluar/src/app/Interface/course.ts
export interface StudentDegree {
  studentId: string;
  studentName?: string | null;
  degree: number;
}

export interface Course {
  id: number;
  crs_Name?: string | null;
  crs_Description?: string | null;
  duration: number;
  departmentId?: number | null;
  departmentName?: string | null;
  studentDegrees: StudentDegree[];
}

File: ITI_API/ITI_API/Model/Course.cs
namespace ITI_API.Model
{
	public class Course
	{
		public int Id { get; set; }
		public string? Crs_Name { get; set; }
		public string? Crs_Description { get; set; }
		public int Duration { get; set; }
		public int? DepartmentId { get; set; }
		public bool IsDeleted { get; set; } = false;
		public virtual Department? Department { get; set; }
		public virtual ICollection<StudentCourse> StudentCourses { get; set; } = new HashSet<StudentCourse>();
	}
}
